### PR TITLE
storage: Add CreateDir

### PR DESCRIPTION
**Pull request by JinnyYi**
ref: https://github.com/beyondstorage/go-storage/issues/560

### CODE REVIEW

File: storage.go [hunk 1]
```diff
@@ -36,20 +36,81 @@ func (s *Storage) completeMultipart(ctx context.Context, o *Object, parts []*Par
...
+	// Add `/` at the end of `path` to simulate a directory.
+	// ref: https://cloud.tencent.com/document/product/436/13324
+	rp += "/"
+
+	putOptions := &cos.ObjectPutOptions{
+		ObjectPutHeaderOptions: &cos.ObjectPutHeaderOptions{
+			ContentLength: 0,
+		},
+	}
+	if opt.HasContentMd5 {
```
Comment (Xuanwo) [inline storage.go]: `CreateDir` doesn't need `content-md5` maybe?

File: storage.go [hunk 2]
```diff
@@ -359,7 +428,11 @@ func (s *Storage) stat(ctx context.Context, path string, opt pairStorageStat) (o
 	o = s.newObject(true)
 	o.ID = rp
 	o.Path = path
```
Comment (Xuanwo) [inline storage.go:429]: We need to trim the `Path` trailing `/`

File: tests/storage_test.go [hunk 3]
```diff
@@ -13,3 +13,19 @@ func TestStorage(t *testing.T) {
 	}
 	tests.TestStorager(t, setupTest(t))
 }
+
+// FIXME: For `CompleteMultipartUpload`, the numbers of the uploaded parts must be continuous and the part information entries in the request body must be sorted by number in ascending order
```
Comment (Xuanwo) [inline tests/storage_test.go]: Let's update tests intead.

File: storage.go (hunk 2 above)
Comment (JinnyYi) [reply storage.go:429]: I’m confused about it. Is `o.Path` not the `path` passed in?

Comment (Xuanwo) [reply storage.go:429]: Yes, in most time `o.Path == path`.

~~But for dir object, we add `/` after path `test` and send real API to `test/`. So after we got the information, we need to roll it back.~~

~~Maybe we could make it more clear by creating a new var: `sp := path + "/"`.~~

In order to make this behavior works as expected, we can add check in `go-integration-test`, make sure returning `path` equals to input `path`.

---

Ooooops, the path hasn't been updated. Current logic is correct.

File: tests/storage_test.go (hunk 3 above)
Comment (JinnyYi) [reply tests/storage_test.go]: Already updated.
ref: https://github.com/beyondstorage/go-service-cos/issues/30

**Review by Xuanwo — APPROVED** [review]

**Pull request merged**